networkx/generators/geometric.py

```diff
@@ -292,9 +292,9 @@ def geographical_threshold_graph(
 
        (w_u + w_v)h(r) \ge \theta
 
-    where `r` is the distance between `u` and `v`, h(r) is a probability of
-    connection as a function of `r`, and :math:`\theta` as the threshold
-    parameter. h(r) corresponds to the p_dist parameter.
+    where `r` is the distance between `u` and `v`, h(r) is any function of
+    `r`, and :math:`\theta` as the threshold parameter. h(r) corresponds to
```

Comment:
It's tricky to get the wording right, but IMO it'd be good to get a little more info here about what `h(r)` "does". The original wording described how `h(r)` should be interpreted (even though that interpretation was not strictly correct). It'd be great if we could give the users some intuition for how `h(r)` should be interpreted; maybe something like

"h(r) gives the relative likelihood of connection as a function of distance, `r`".

Of course, "likelihood" is a loaded term too :).

This is just a thought - if it's too hard to come up with something I think the formula above is adequate to illustrate the purpose of `h(r)`.

Reply:
Thanks for the suggestions. I'll get those implemented later today. One minor thing I was thinking about is why use 'h(r)' at all? It doesn't show up anywhere else and it's a little confusing to switch back and forth between p_dist and h(r). Instead using something like 'p(r)' may be more intuitive to the typical user. (I did take a look at the paper in ref 1 and they use h(r), but I imagine a person going from the paper to the implementation wouldn't have trouble with that switch).

Reply:
That's a good point, +1 for getting rid of the extra mapping between `h` and `p_dist`... the docstring should really correspond more to the implementation than the original paper (though a bullet could be added to the `Notes` section about the equivalence of `p_dist` and `h(r)` from the original paper).

Reply:
Cool. I updated things and it should be a bit more clear now.

Reply:
Now that I think more about it. Why have an argument named `p_dist` at all? It kind of adds to the confusion because it doesn't need to be a probability distribution. I think a name like `h` instead of `p_dist` would be more clear.

Reply:
>Now that I think more about it. Why have an argument named p_dist at all? It kind of adds to the confusion because it doesn't need to be a probability distribution. I think a name like h instead of p_dist would be more clear.

Though I agree with your reasoning, unfortunately this is a backwards-incompatible change and thus would require a `FutureWarning` to notify users of the name change. If we want to go this route, I would advocate for:
 - Fix up the docstring in this PR, leaving out the argument name change
 - Submit a separate PR with the proposed name change so we can keep all of the warning/user-notification & related discussion self-contained.